src/components/views/auth/InteractiveAuthEntryComponents.tsx

```diff
@@ -786,27 +811,31 @@ export class FallbackAuthEntry extends React.Component {
         }
         return (
             <div>
-                <a href="" ref={this._fallbackButton} onClick={this._onShowFallbackClick}>{ _t("Start authentication") }</a>
+                <a href="" ref={this.fallbackButton} onClick={this.onShowFallbackClick}>{
+                    _t("Start authentication")
+                }</a>
                 {errorSection}
             </div>
         );
     }
 }
 
-const AuthEntryComponents = [
-    PasswordAuthEntry,
-    RecaptchaAuthEntry,
-    EmailIdentityAuthEntry,
-    MsisdnAuthEntry,
-    TermsAuthEntry,
-    SSOAuthEntry,
-];
-
-export default function getEntryComponentForLoginType(loginType) {
-    for (const c of AuthEntryComponents) {
-        if (c.LOGIN_TYPE === loginType || c.UNSTABLE_LOGIN_TYPE === loginType) {
-            return c;
-        }
+export default function getEntryComponentForLoginType(loginType: AuthType): typeof React.Component {
+    switch (loginType) {
```

Comment:
if we're not using the `LOGIN_TYPE` static member anymore we can probably purge it.

Reply:
At the moment, it's nearly unused, except for the ones on `SSOAuthEntry` and `PasswordAuthEntry`, in blocks such as these:

https://github.com/matrix-org/matrix-react-sdk/blob/44ecf62758f51d752ec1b99121402e4bb177090a/src/AddThreepid.js#L193-L217
https://github.com/matrix-org/matrix-react-sdk/blob/44ecf62758f51d752ec1b99121402e4bb177090a/src/components/views/dialogs/DeactivateAccountDialog.js#L53-L75

Actually I initially removed the `LOGIN_TYPE`s and converted the above to import the `AuthType` enum to replace it... but then I saw I'd still need `SSOAuthEntry` for `PHASE_PREAUTH` etc. and then I wasn't sure what would be best any more.

Opinions?

Reply:
Slowly bringing it to death like this is probably fine, given its usage. One day we'll clean up the rest and it'll get refactored down as needed.